pkg/openthread/contrib/conn/conn_udp.c

```diff
@@ -0,0 +1,101 @@
+#include "net/conn/udp.h"
+#include "conn.h"
+#include "stdio.h"
+#include <openthread.h>
+#include "string.h"
+#include <net/ipv6/addr.h>
+#include "ot.h"
+
+static msg_t _msg;
+
+static conn_udp_msg_t msg_info;
+static otSockAddr sockaddr = {};
+
+void HandleUdpReceived(void *aContext, otMessage aMessage, const otMessageInfo *aMessageInfo)
+{
+	msg_info.message = aMessage;
+	msg_info.message_info  = (otMessageInfo*) aMessageInfo;
+	_msg.type = OPENTHREAD_MSG_TYPE_RECV;
+	_msg.content.ptr = (void*) &msg_info;
+	conn_udp_t *conn = (conn_udp_t*) aContext;
+	msg_send(&_msg, conn->receiver_pid);	
+}
+
+int conn_udp_create(conn_udp_t *conn, const void *addr, size_t addr_len, int family, uint16_t port)
+{
+	memcpy(&sockaddr.mAddress.mFields, addr, addr_len);
+    sockaddr.mPort = port;
+
+    otOpenUdpSocket(&conn->mSocket, &HandleUdpReceived, conn);
+    otBindUdpSocket(&conn->mSocket, &sockaddr);
+    return 0;
+}
+
+int conn_udp_sendto(const void *data, size_t len, const void *src, size_t src_len,
+                    const void *dst, size_t dst_len, int family, uint16_t sport,
+                    uint16_t dport)
+
+	if(dst_len != sizeof(ipv6_addr_t) || src_len != sizeof(ipv6_addr_t))
+		return -EINVAL;
+
+    otMessage message;
+	otUdpSocket mSocket;
+
+    message = otNewUdpMessage();
+    otSetMessageLength(message, len);
+    otWriteMessage(message, 0, data, len);
+
+	otMessageInfo mPeer;
+
+	//Set source address
+	memcpy(&mPeer.mSockAddr.mFields, src, src_len);
+
+	//Set source port
+	mSocket.mSockName.mPort = sport;
+
+	//Set dest address
+	memcpy(&mPeer.mPeerAddr.mFields, dst, dst_len);
+
+	//Set dest port
+    mPeer.mPeerPort = dport;
```

Comment:
Yes, I will add the proper comments/copyright and run uncrustify to fix these :)